packages/compass-connections/src/stores/connections-store-redux.ts

```diff
@@ -1507,6 +1508,46 @@ function isAtlasStreamsInstance(
   }
 }
 
+const NonRetryableErrorCodes = [3000, 3003, 4004, 1008] as const;
```

Comment:
Maybe worth linking to the code in mms / expanding the description here a bit so that the context for where those codes are coming from is not lost and easy to link to the source